LinkedListHomework9/Program.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.Xml.Linq;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace LinkedListHomework9
+{
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            // Task 1
+            Console.WriteLine("Task 1");
+            int[] numbers = { 2, 4, 3, 2, 8, 2, 5, 1, 2 };
+            LinkedList<int> numbersLinkedList = new LinkedList<int>(numbers);
+            InsertNodesAfterSelectedValue(numbersLinkedList, 2, 10);
+
+            // Task 2
+            Console.WriteLine("\nTask 2");
+            int[] list1 = { 2, 4, 34, 2, 8, 25, 5, 106, 2 };
+            int[] list2 = { 2, 4, 8, 77, 8, 2, 5, 109, 2 };
+            LinkedList<int> numbersLinkedList1 = new LinkedList<int>(list1);
+            LinkedList<int> numbersLinkedList2 = new LinkedList<int>(list2);
+            Console.WriteLine($"Common values between 2 linked lists:");
+            var commonValuesLinkedList = GetCommonValuesFromTwoLinkedLists(numbersLinkedList1, numbersLinkedList2);
+            OutPutLinkedListValues(commonValuesLinkedList);
+        }
+
+        /* Task 1. Создайте LinkedList и два элемента, вставьте второй элемент
+         * после каждого вхождения первого элемента в списке. https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.linkedlist-1?view=net-6.0
+         * Итак, если список равен [2,4,3,2,8,2,5,1,2], а элементы равны 2 и 10,
+         * результатом будет [2,10,4,3,2,10,8,2,10,5,1,2,10] */
+        private static void InsertNodesAfterSelectedValue(LinkedList<int> linkedList, int firstElement, int secondElement)
+        {
+            var changedList = linkedList.ToList();
+            LinkedList<int>.Enumerator em = linkedList.GetEnumerator();
+            var indexHelper = 0;
+            while (em.MoveNext())
```

Comment:
JFYI - Another solution for LinkedList:

            var currentNode = linkedList.First;

            while (currentNode != null)
            {
                if (currentNode.Value == firstElement)
                {
                    linkedList.AddAfter(currentNode, new LinkedListNode<int>(secondElement));
                    currentNode = currentNode.Next;
                }

                currentNode = currentNode.Next;
            }

Reply:
Just close this comment and merge

Reply:
Much better!  🤔  Thank you!